model-optimizer/extensions/middle/ApplyPermutations.py

```diff
@@ -143,6 +144,9 @@ def permute_input_data(graph: Graph):
                         not is_input_data_in_correct_layout(node, in_port) and \
                         len(port_to_check.data.get_shape()) >= 4:
                     permutation(node, port_info, in_port)
+            if node.has_and_set('need_shape_inference'):
+                node.infer(node)
+                node.need_shape_inference = False
```

Comment:
Instead of explicit call of for shape inference, try to use `force_shape_inference=True`

Reply:
Also, it is better to refactor this transformation to use new API.

Reply:
Don't think we should do any refactoring in this commit